Sources/StreamChat/Controllers/EntityDatabaseObserver.swift

```diff
@@ -133,7 +137,11 @@ class EntityDatabaseObserver<Item, DTO: NSManagedObject> {
             cacheName: nil
         )
         
-        _item.computeValue = { [unowned self] in
+        _item.computeValue = { [weak self] in
+            // Ideally, this should rather be `unowned`, however, `deinit` is not always called on the same thread as this
+            // callback which can cause a race condition when the object is already being deinited on a different thread.
+            guard let self = self else { return nil }
```

Comment:
We don't need `[weak self]` here, we can get by with `[weak frc]` and the `guard let fetchedObjects = frc?.fetchedObjects` will take care of it

Reply:
Correct 👍

Reply:
@b-onc whad do you think about 320a228 ?